### PR TITLE
Fix numeric_iso_code in the upgrade to 1.7.7.1

### LINKED ISSUE #22208: Editing of deleting a currency KO after upgrade from 176 to 1770 if currency was modified in 176
#### Describe the bug

When a currency was modified in 176, if I upgrade to 1770, the currency can no longer be modified or deleted in 1770 BO, an error is displayed. There is also an error if I try to upload again the currency using the "localization pack".

According to investigation with some devs, it's linked to the numeric iso which is saved as "12" after edit, but should be at "012". That explains why this bug only happens on currencies with numeric iso code < 100, such as Algerian Dinar (code 012) or Argentine Pesos (code 032).

See Wikipedia page for numeric iso code of currencies : 

https://en.wikipedia.org/wiki/ISO_4217#Active_codes

#### Expected behavior

Numeric iso code should be saved on 3 digits and not 2.

#### Steps to Reproduce

1. Install 1.7.6.x version
2. In BO, add localization pack (for example, Algeria) in order to have the DZD currency
3. Edit the currency (for example, change exchange rate)
4. Upgrade to 1.7.7.0
5. Go to BO > International > Localization > Currencies > Edit DZD currency : The following error is displayed : 
`An unexpected error occurred. [PrestaShop\PrestaShop\Core\Domain\Currency\Exception\CurrencyException code 0]: An error occurred when deleting Currency object with id "2"`
6. Try to delete the currency : Same error
7. Go to "Localization" tab, try to add Algeria Pack : An exception is displayed `Property Currency->numeric_iso_code is not valid
`

**Screenshots**

Screenrecord of behavior after upgrade : 

https://drive.google.com/file/d/1jID2MoVJbeTcK0TaR3q8IVi4B47Md1fP/view?usp=sharing

Screenshot of exception : 

edit currency after upgrade 2

Screenshot of the database : Value for DZD is "12", should be "012"

edit currency after upgrade db 1

#### Additional information

* PrestaShop version: 1.7.7.0
* PHP version: N/A

### PR DESCRIPTION
**Pull request by atomiix**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Since 1.7.7.0, currency's `numeric_iso_code` should be 3 characters long. This PR aims to update older `numeric_iso_code` that were 2 characters long by prepending them with a '0' during the upgrade process.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #22208
| How to test?  | 1. Install a 1.7.6 shop<br>2. Add the DZD currency<br>3. Either make a 1.7.7.1 build with this PR and upgrade to it or Checkout the 1.7.7.x branch with this PR, update `_PS_INSTALL_VERSION_` in `install-dev/install_version` to `1.7.7.1` and run `php install-dev/upgrade/upgrade.php`<br>4. You should be able to edit/delete the DZD currency without any error

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22214)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

**Review by eternoendless — DISMISSED** [review]

File: install-dev/upgrade/sql/1.7.7.1.sql [hunk 1]
```diff
@@ -1 +1,3 @@
 /* PHP:ps_1771_update_customer_note(); */;
+
+UPDATE `PREFIX_currency` SET numeric_iso_code = CONCAT('0', numeric_iso_code) WHERE LENGTH(numeric_iso_code) = 2;
```
Comment (PierreRambaud) [inline install-dev/upgrade/sql/1.7.7.1.sql]:
```suggestion
UPDATE `PREFIX_currency` SET numeric_iso_code = LPAD(numeric_iso_code, 3, '0') WHERE LENGTH(numeric_iso_code) != 3;
```
What about numeric_iso_code before 10 :trollface:

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
One thing we missed

**Review by PierreRambaud — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM !!
Versions tested ( multiple time):
from 1.7.6.9 => php 7.1 , mysql 5.7
from 1.7.6.9 => php 7.1 , mariadb 10.1
from 1.7.6.5 => php 7.1 , mysql 5.7

Thanks @atomiix !!

**Comment by Progi1984** [comment]
Thanks @atomiix & @SD1982

**Pull request merged**